tests/common/Language/Java/InlineSpec.hs

```diff
@@ -122,3 +128,27 @@ spec = do
           st :: J ('Class "java.lang.Thread$State") <-
             [java| Thread.State.NEW |]
           [java| $st == Thread.State.NEW |] `shouldReturn` True
+
+      it "Supports modified utf-8 encoding from Haskell to Java" $
+          withLocalRef (reflect ("a\NULb" :: T.Text)) $ \jString ->
+            [java| new String("a\0b").equals($jString) |] `shouldReturn` True
```

Comment:
How about
```suggestion
          withLocalRef [java| "a\0b" |] reify `shouldReturn` ("a\NULb" :: T.Text)
```
?
Besides being shorter, I guess the error message would be more telling in case of failure.

Reply:
Ah, nevermind. I see now that you want to test the other direction of the encoding.